sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -49,8 +49,10 @@ trait FunctionRegistryBase[T] {
 
   final def registerFunction(
       name: FunctionIdentifier, builder: FunctionBuilder, source: String): Unit = {
+    var builderClassName = builder.getClass.getCanonicalName
```

Comment:
Java 11+ has `Objects.requireNonNullElse(obj, defaultObj)`.

For Java 8, maybe we can use `MoreObjects.firstNonNull`, then `builderClassName` can be a `val`

Reply:
I adopted the Option one liner suggested by @zhenlineo